jena-arq/src/main/java/org/apache/jena/query/DatasetFactory.java

```diff
@@ -16,253 +16,226 @@
-     * @return Dataset
-     */
-
-    public static Dataset create(List<String> uriList, List<String> namedSourceList, String baseURI) {
-        return DatasetUtils.createDataset(uriList, namedSourceList, baseURI) ;
-    }
-
-    public static Dataset make(Dataset ds, Model defaultModel) {
-        Dataset ds2 = new DatasetImpl(ds) ;
-        ds2.setDefaultModel(defaultModel) ;
-        return ds2 ;
-    }
-
-    // Assembler.
-    /**
-     * Assembler a dataset from the model in a file
-     * 
-     * @param filename
-     *            The filename
-     * @return Dataset
-     */
-    public static Dataset assemble(String filename) {
-        Model model = FileManager.get().loadModel(filename) ;
-        return assemble(model) ;
-    }
-
-    /**
-     * Assembler a dataset from the model in a file
-     * 
-     * @param filename
-     *            The filename
-     * @param resourceURI
-     *            URI for the dataset to assembler
-     * @return Dataset
-     */
-    public static Dataset assemble(String filename, String resourceURI) {
-        Model model = FileManager.get().loadModel(filename) ;
-        Resource r = model.createResource(resourceURI) ;
-        return assemble(r) ;
-    }
-
-    /**
-     * Assembler a dataset from the model
-     * 
-     * @param model
-     * @return Dataset
-     */
-    public static Dataset assemble(Model model) {
-        Resource r = GraphUtils.findRootByType(model, DatasetAssembler.getType()) ;
-        if ( r == null )
-            throw new ARQException("No root found for type <" + DatasetAssembler.getType() + ">") ;
-
-        return assemble(r) ;
-    }
-
-    /**
-     * Assembler a dataset from a resource
-     * 
-     * @param resource
-     *            The resource for the dataset
-     * @return Dataset
-     */
-
-    public static Dataset assemble(Resource resource) {
-        Dataset ds = (Dataset)Assembler.general.open(resource) ;
-        return ds ;
-    }
 
+	/**
+	 * @return an in-memory, modifiable Dataset
+	 * @deprecated Prefer {@link #createGeneral()} or {@link #createTxnMem()}
+	 */
+	@Deprecated
+	public static Dataset createMem() {
+		return create(DatasetGraphFactory.createMem());
+	}
+
+	/**
+	 * @return a transactional, in-memory, modifiable Dataset with MR+SW locking
+	 */
+	public static Dataset createTxnMem() {
+		return create(DatasetGraphFactory.createTxMem());
+	}
+
+	/**
+	 * @return a general-purpose, in-memory, modifiable Dataset
+	 */
+	public static Dataset createGeneral() {
+		return createTxnMem();
+	}
+
+	/**
+	 * @return an in-memory, modifiable Dataset. New graphs must be explicitly added using .addGraph.
+	 */
+	public static Dataset createMemFixed() {
+		return create(DatasetGraphFactory.createMemFixed());
+	}
+
+	/**
+	 * @param model The model for the default graph
+	 * @return a dataset with the given model as the default graph
+	 */
+	public static Dataset create(final Model model) {
+		return new DatasetImpl(model);
```

Comment:
Why adding al the `final`?  The compiler, with "effectively final", does this anyway.

Adding `final` to arguments just increases the syntax line noise. A lot of this file has changed for nu functionality change.